### PR TITLE
GroupController - initial implementation

### PR DESCRIPTION
**Pull request by annapawlas**

### CODE REVIEW

File: src/main/java/com/kodilla/ecommercee/controller/GroupController.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    @GetMapping
+    public ResponseEntity<List<GroupDto>> getGroups(){
+      return ResponseEntity.ok(groupDtoList);
+    }
+
+    @GetMapping(value="{groupId}")
+    public ResponseEntity<GroupDto> getGroup(@PathVariable Long groupId){
+        for (GroupDto group : groupDtoList) {
+            if (group.getGroupId().equals(groupId)){
+                ResponseEntity.ok(group);
```
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/controller/GroupController.java]: you missed "return" statement - should be 
`return ResponseEntity.ok(group)`
without it, only empty ok().build() is returned

File: src/main/java/com/kodilla/ecommercee/controller/GroupController.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<GroupDto> updateGroup(@RequestBody GroupDto groupDto) {
+        for (GroupDto group : groupDtoList) {
+            if (group.getGroupId().equals(groupDto.getGroupId())){
+                group.getName().replaceFirst(group.getName(), groupDto.getName());
+            }
+        }
+        return ResponseEntity.ok().build();
```
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/controller/GroupController.java]: As above - no content is returned, only empty "ok".
Here you can do it e.g. like this:
`return ResponseEntity.ok(groupDto);`

**Review by WorstWorm — COMMENTED** [review]
- Think if it would not be good to add a response in case of an incorrect call - e.g. trying to update a non-existent group, etc.
- The question is whether we put the DTO to the DTO package or, as in your commit, to the domain package. I assume that the classes User, Product, etc. will go to the domain also. Wouldn't it be better if the DTOs were separate? I think it will be more readable.

File: src/main/java/com/kodilla/ecommercee/controller/GroupController.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/groups")
+@RequiredArgsConstructor
```
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/controller/GroupController.java]: What is the use of `@RequiredArgsConstructor` here?
GroupController has no atributes.

**Review by WorstWorm — APPROVED** [review]
All :+1:

**Review by BartoszSzych98 — APPROVED** [review]

**Pull request merged**